### DISCUSSION TITLE
Change theme for SSH session without using current ssh kitten (connection issues on non-unix machines).

### DISCUSSION THREAD

**Discussion by mwpardue** [kovidgoyal/kitty#6316] [Q&A]
Hi, so I make pretty heavy use of the ssh kitten for Linux servers, and IMO one of the greatest features of that kitten is the ability to change the theme for the duration of an ssh session. This helps to differentiate the machines between production, sandbox, customerA, customerB, etc. at a glance. However, I am unable to do this same thing for non-Linux machines, specifically network hardware like switches and firewalls. When I try to connect to those machines via the ssh kitten I get various errors, some machines give me an error related to ssh command execution not being enabled and some, like Cisco ASAs, just spit out a random string of base64 characters before failing to connect. I'm sure this is related to the environment copying, and the kitten is doing precisely what it's supposed to do. 

Is there a way I can launch the kitten so it runs just like a "normal" ssh session, in addition to changing the theme for the duration of the remote session? Or is there another way to achieve this desired behavior other than the ssh kitten? I'm prepared to try my hand at writing my own kitten, but wanted to check here first. 

Thanks for any suggestions anybody can provide!

**Comment by kovidgoyal** [comment]
Your best bet is to use the interpreter option assuming you have python
installed on these machines, use that as the interpreter. If not then
yes you would need to write your own kitten. The ssh kitten assumes you
have a working POSIX env or Python on the remote machine. Someday I may
write a tiny dsl which will allow you to use the kitten binary itself as
an interpreter, but that's not a high priority.

**Reply by mwpardue** [reply]
No, these devices don't have a working python installation, or at least using the python interpreter doesn't work. I'll look at building my own kitten to achieve my goal. I think I can use the current kitten as a model and it'll probably make the whole thing much easier. 

Thanks for the fast response!!
